CONTRIBUTING.md

```diff
@@ -97,6 +97,17 @@ For that reason, command arguments tend to match the path and query params of th
 with the last param being a required argument and the rest handled via flag options. 
 For commands that create or modify complex data structures, the use of configuration files is preferred over flag options.
 
+#### How to define flags:
+Flags are a way to modify the command, also may be called "options".  Flags always have a long version with two dashes (--state) but often also have a shortcut with one dash and one letter (-s).
+
+`mongocli` uses the following types of flags:
+
+- `--flagName value`: this type of flag passes the value to the command. Examples: `--projectId 5efda6aea3f2ed2e7dd6ce05`
+- `--booleanFlag`: this flag represents a boolean and it sets the related variable to true when the flag is used, false otherwise.  Example: `--force`
+- `--flagName value1,value2,..,valueN`: you will also find flags that accept a list of values. This type of flag can be very useful to represent data structures as `--role roleName1@db,roleName2@db` or `--privilege action@dbName.collection,action2@dbName.collection`. 
```

Comment:
[nit] we don't always use `@`, roles and privilege just happen to have a standard format with `@`
meaning this role/privilege at this collection/db, we usually use `:` to split structs as well

Reply:
Shouldn’t we decide on only one format and stick with it?

Reply:
> Shouldn’t we decide on only one format and stick with it?

Maybe? can you do an inventory and see what have we been doing? `@` for denoting "at", makes sense but not sure what else we have have out there and if it would. going for only `@` would also limit the usage to only things that have that kind of relationship

Reply:
We have only the following commands that use a similar structure:

- `atlas|om dbuser create/update --role roleName@dbName --scope resourceName:resourceType`
- `atlas dbrole(s) create/update --privilege action@dbName.collection`
- `om cluster index create --key field:type`

Personally, I am fine with either approach (`:` or `@`) for splitting the first and the second values but `@` required less changes

Reply:
@gssbzn

Reply:
that `field:type` would be weird with an `@` like you say Ii'm ok to limit to those two and decide when it makes sense to use one or the other

Reply:
updated